File: Programming Skills II/896. Monotonic Array/896. Monotonic Array.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var isMonotonic = function(nums) {
    let dif = 0;
    for(let i=0; i<nums.length-1; ++i){
        if( nums[i] != nums[i+1] && dif == 0 )
            dif = nums[i] - nums[i+1];
        else if( (nums[i] > nums[i+1] && dif < 0 ) || (nums[i] < nums[i+1] && dif > 0 ) )
            return false;
    }
    return true;
};

File: Programming Skills II/896. Monotonic Array/896. Monotonic Array.cpp
class Solution {
public:
    bool isMonotonic(vector<int>& nums) {
        int n = nums.size(), dif = 0;
        for(int i=0; i<n-1; ++i){
            if(nums[i] != nums[i+1] && dif == 0 )
                dif = nums[i] - nums[i+1]; 
            else if( (nums[i] > nums[i+1] && dif < 0)  || (nums[i] < nums[i+1] && dif > 0) )
                return false;
        }
        return true;
    }
};
